wordpress-mobile/gutenberg-mobile
# Improving accessibility on Post Title

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/912

Changes reside in gutenberg PR: https://github.com/WordPress/gutenberg/pull/15106

This PR improves accessibility (screen reader) for the Post Title. In particular when it's not focused.

Empty title:

**Label:** "Post title. Empty"

Title with a text:

**Label:** "Post title. [the text]"

When caret is blinking on Post Title the accessibility behavior is handed over to the OS and the system decides how to handle the accessibility for the text input field.

**To test (iOS):**

Activate VoiceOver: https://github.com/wordpress-mobile/WordPress-iOS/wiki/Using-VoiceOver

**To test (Android):**

Activate TalkBack: https://www.wikihow.com/Enable-TalkBack-on-Your-Android
In my case, the system had a handy tutorial on how to use it.
If your version doesn't have it, it's actually very similar to VoiceOver.

**Test Steps**
Tap on the title when it is empty
It should say: "Post title. Empty"

Double tap to enter some text and focus to another text input
Tap on the title again
It should say: "Post title. [entered text]"

**A note on Android**

On Android I could only be able to hear "Post title. Empty" when I tap the area out of the text input. It looks like childs continue to be accessible and I am not sure this is sth that can be resolved.
